server/models/service-process.js

```diff
@@ -43,7 +44,10 @@ module.exports = function extendServiceProcess(ServiceProcess) {
         return proc.updateAttributes({
           stopTime: new Date(),
           stopReason: pInfo.reason,
-        }, asyncCb);
+        }, function(err, proc) {
+          if (err) return asyncCb(err);
+          serviceManager.onProcessExit(proc, asyncCb);
```

Comment:
I don't see the relationship between any of the service-process changes and gateway, are they an unrelated bug fix?